enhancements/sig-policy/72-content-to-ansilbe-job/README.md

```diff
@@ -0,0 +1,188 @@
+
+- `namespace` is the namespace of the root policy, which is from `rootPolicy.GetNamespace()`
+
+- `hubcluster` is the name of the hub cluster, which is from `rootPolicy.GetClusterName()`
+
+- `policy_violation_context` is a map. The keys are non-compliant cluster names, and the value is the raw replicated 
+ policy status for each non-compliant cluster.
+ The policy automation looks at the `cluster namespaces` of the hub cluster and gets the replicated policies. 
+ Then it filters out the non-compliant replicated policies and puts the status into the `policy_violation_context` map.
+
+ Suppose the policy automation runs on the hub cluster without the managed cluster. In that case, `policy_violation_context` 
+ would be an empty map.
+ 
+ The field path of the raw status is `policy.policyStatus`. The below unnecessary fields in `policy.policyStatus` will be \
+ skipped and not in `policy_violation_context`:
+ 1. `policy.policyStatus.details[n].templateMeta`.
+ 2. `policy.policyStatus.details[n].history[n].eventName`
+
+### User Stories
+
+#### Story 1
+
+As an OCM Policy add-on user, I would like to get the additional status in an Ansible job's `extra_vars` field. For example,
+regarding the CertificatePolicy policy, which checks for expiring certificates in secrets, I want to get the replicated policy 
+status for each non-compliant cluster so I can know which secret expires on a specific cluster in the Anisble Job. Then, the 
+Ansible Automation Platform can create a notice ticket and provide the violation details. 
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigation
+
+Since `policy_violation_context` is the replicated policy status for all non-compliant managed clusters, the length of 
+`policy_violation_context` may be oversized when the amount of managed clusters is large.
+
+To solve this, we will only pass the top `N` replicated policy status into `policy_violation_context`, where `N` is configurable
+via an optional policy automation field called `policyViolationContextLimit`. If this field is not set, `N` will default to the
+first 1000 managed clusters.
+
+## Design Details
+
+See the [Proposal](#proposal) section.
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+- Unit tests can be implemented to check the created Ansible job. For each scenario, the `extra_vars` contain the designed data.
+
+### Graduation Criteria
+
+1. The proposal is reviewed and accepted
+2. Implementation is completed to support the primary goals
+3. Test cases developed to demonstrate that the user stories are fulfilled
+
+### Upgrade / Downgrade Strategy
+
+There are no concerns with upgrades since this is not a breaking change.
+
+Downgrading would only pass `target_clusters` to Ansible Job, which is an acceptable outcome.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+N/A
+
+## Drawbacks
+
+1. This design does not restructure `policy_violation_context` data for different policies.
```

Comment:
what's the "different policies" ?  different type policies ?  configuration and cert in one policy ?

Reply:
For different types of policies, we can parse the violation message from each replicated policy status. 

For example, for certificate policy, we can create the below object based on the parsed violation messages. This object is easier to use on the Ansible Tower. In the current scope, we decide to just pass the raw replicated policy status rather than parsing the violation messages.

{
    "cluster1": {
        "CertificatePolicy/certificate-policy-name": {
            "noncompliant_objects": [
                {
                    "name": "secret1",
                    "namespace": "my-project-namespace",
                    "kind": "Secret"
                },
                {
                    "name": "secret2",
                    "namespace": "my-project-namespace",
                    "kind": "Secret"
                },
                {
                    "name": "super-secret",
                    "namespace": "my-other-namespace",
                    "kind": "Secret"
                }
            ]
        }
    }
}